network/network_linux.go

```diff
@@ -139,6 +148,88 @@ func (nm *networkManager) saveIPConfig(hostIf *net.Interface, extIf *externalInt
 	return err
 }
 
+func getMajorVersion(version string) (int, error) {
+	versionSplit := strings.Split(version, dotDelimiter)
+	if len(versionSplit) > 0 {
+		retrieved_version, err := strconv.Atoi(versionSplit[0])
+		if err != nil {
+			return 0, err
+		}
+
+		return retrieved_version, err
+	}
+
+	return 0, fmt.Errorf("[net] Error getting major version")
+}
+
+func isGreaterOrEqaulUbuntuVersion(versionToMatch int) bool {
+	osInfo, err := platform.GetOSDetails()
+	if err != nil {
+		log.Printf("[net] Unable to get OS Details: %v", err)
+		return false
+	}
+
+	log.Printf("[net] OSInfo: %+v", osInfo)
+
+	version := osInfo[versionID]
+	distro := osInfo[distroID]
+
+	if strings.EqualFold(distro, ubuntuStr) {
+		version = strings.Trim(version, "\"")
+		retrieved_version, err := getMajorVersion(version)
+		if err != nil {
+			log.Printf("[net] Not setting dns. Unable to retrieve major version: %v", err)
+			return false
+		}
+
+		if retrieved_version >= versionToMatch {
+			return true
+		}
+	}
+
+	return false
+}
+
+func readDnsServerIP(ifName string) (string, error) {
+	cmd := fmt.Sprintf("systemd-resolve --status %s", ifName)
+	out, err := platform.ExecuteCommand(cmd)
+	if err != nil {
+		return "", err
+	}
+
+	log.Printf("[net] console output for above cmd: %s", out)
```

Comment:
"console output for %s: %s", cmd, out)

Reply:
in previous line, we print the command..so not needed